File: resources/typescript/functions-and-methods.ts
class FunctionsAndMethods {
    constructor() {}

    public testFunction1(x: number, y: string): number {
        return 1;
    }

    testFunction2(x: number, y: string): number {
        return 1;
    }

    private testFunction3(): number {
        return 2;
    }

    public static staticMethod() {}
}

function testMethod() {}

function outOfScope(): number {
    return 3;
}

const myFunction = function myFunctionDefinition() {
    return 42;
};

const x = (x, y) => x * y;

interface KeyValueProcessor{ functionName(key: number, value: string): void; }

interface KeyValueProcessor{ (key: number, value: string): void; };

abstract class Animal {
    abstract makeSound(input : string) : string;
}

File: resources/c-sharp/functions-and-methods.cs
FunctionsAndMethods obj = new FunctionsAndMethods();
Del d = obj.testFunction1;

void testMethod() {}

int outOfScope() {
    return 3;
}

public class FunctionsAndMethods {

    private string hello;

    public FunctionsAndMethods() {
        hello = "hello";
    }

    public int testFunction1(int x, int y) {
        return 1;
    }

    int testFunction2(int x, int y) {
        return 1;
    }

    private int testFunction3() {
        return 1;
    }

    public static void staticMethod() {}

}

public delegate int Del(int x, int y);
